File: backend-model/src/main/java/outcobra/server/model/dto/SchoolYearDto.java
package outcobra.server.model.dto;

import outcobra.server.model.SchoolYear;
import outcobra.server.model.interfaces.MappableDto;
import outcobra.server.model.interfaces.Mapper;

public class SchoolYearDto implements MappableDto<SchoolYearDto, SchoolYear> {
    @Override
    public Mapper<SchoolYearDto, SchoolYear> getMapper() {
        throw new UnsupportedOperationException("not implemented");
    }

    @Override
    public SchoolYear toEntity() {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: backend/src/main/kotlin/outcobra/server/model/dto/SchoolYearDto.kt
package outcobra.server.model.dto

import outcobra.server.model.interfaces.OutcobraDto
import outcobra.server.model.interfaces.UserParentLinkedDto
import java.time.LocalDate

data class SchoolYearDto(
        val id: Long = 0,
        val schoolClassIds: List<Long> = listOf(),
        val name: String = "",
        val validFrom: LocalDate = LocalDate.now(),
        val validTo: LocalDate = LocalDate.now(),
        override var userId: Long = 0,
        val semesters: List<SemesterDto> = arrayListOf()) : UserParentLinkedDto, OutcobraDto {

    override val identifier: Long
        get() = id
}